### PR TITLE
Add debug modes for tensor generate in conv and pool

### PR DESCRIPTION
**Pull request by skrsticTT**
### Ticket
/

### What's changed
Added 2 modes (**random** existed although had no name) in `randomize_torch_tensor` function for easier debugging:
- **stick** : each stick is going to have same value, from 0 and then increasing by 1
- **single** : each datum in tensor is same and you need to provide that value

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/18411500115) ⏳ 
- [ ] [Nightly tt-metal L2 tests](https://github.com/tenstorrent/tt-metal/actions/runs/18411514328) ⏳

### CODE REVIEW

File: tests/ttnn/nightly/unit_tests/operations/pool/test_maxpool2d.py [hunk 1]
```diff
@@ -21,13 +21,32 @@ def tensor_map(request):
...
-        tensor_map[tensor_shape] = torch_tensor
+        if mode == "random":
+            torch_tensor = torch.randn(tensor_shape, dtype=torch.bfloat16)
+        elif mode == "stick":
+            torch_tensor = torch.zeros(tensor_shape, dtype=torch.bfloat16)
+            for n in range(tensor_shape[0]):
+                for c in range(tensor_shape[1]):
+                    for h in range(tensor_shape[2]):
+                        for w in range(tensor_shape[3]):
+                            torch_tensor[n, c, h, w] = h * tensor_shape[3] + w
```
[the remark this answers is not in the document]
Comment (skrsticTT) [reply tests/ttnn/nightly/unit_tests/operations/pool/test_maxpool2d.py]: haha nice, going to do it

File: tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py [hunk 2]
```diff
@@ -58,21 +58,44 @@ def torch_tensor_map(request):
...
-            torch_tensor_map[tensor_shape] = torch_tensor
+            if generate_positive_numbers:
+                torch_tensor = torch.abs(torch_tensor)
+        elif mode == "stick":
+            torch_tensor = torch.zeros(tensor_shape, dtype=torch.bfloat16).float()
+            for n in range(tensor_shape[0]):
+                for c in range(tensor_shape[1]):
+                    for h in range(tensor_shape[2]):
+                        for w in range(tensor_shape[3]):
+                            torch_tensor[n, c, h, w] = h * tensor_shape[3] + w
```
[the remark this answers is not in the document]
Comment (skrsticTT) [reply tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py]: +1

File: tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py [hunk 3]
```diff
@@ -58,21 +58,44 @@ def torch_tensor_map(request):
...
+            if generate_positive_numbers:
+                torch_tensor = torch.abs(torch_tensor)
```
[the remark this answers is not in the document]
Comment (skrsticTT) [reply tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py]: makes sense

**Review by dstoiljkovicTT — APPROVED** [review]

File: tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py [hunk 4]
```diff
@@ -58,21 +58,39 @@ def torch_tensor_map(request):
...
+            h, w = tensor_shape[2], tensor_shape[3]
+            stick = torch.arange(h * w, dtype=torch.bfloat16).reshape(1, 1, h, w)
+            torch_tensor = stick.expand(tensor_shape)
+        elif mode == "single":
+            if fill_value is None:
+                raise ValueError("fill_value must be provided when mode is 'single'")
+            torch_tensor = torch.full(tensor_shape, fill_value, dtype=torch.bfloat16).float()
+        else:
+            raise ValueError(f"Unsupported mode: {mode}. Use 'random', 'stick', or 'single'")
+
```
Comment (s-jovic) [inline tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py:96]: Can we reuse this between convs and pools?

File: tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py [hunk 5]
```diff
@@ -58,21 +58,39 @@ def torch_tensor_map(request):
...
-            torch_tensor = torch_tensor_map[tensor_shape]
-        else:
+        if mode == "random":
             torch_tensor = torch.randn(tensor_shape, dtype=torch.bfloat16).float()
-            torch_tensor_map[tensor_shape] = torch_tensor
+            if generate_positive_numbers:
+                torch_tensor = torch.abs(torch_tensor)
+        elif mode == "stick":
+            h, w = tensor_shape[2], tensor_shape[3]
+            stick = torch.arange(h * w, dtype=torch.bfloat16).reshape(1, 1, h, w)
```
Comment (nilaykp) [inline tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py:88]: This can saturate in case, we use higher dims, towards end of image(may be multiplying with 0.1 might help or something else). what do you think??.

File: tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py (hunk 4 above)
Comment (skrsticTT) [reply tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py:96]: maybe we can but i do not see any 'common' place, do you have any suggestion?

File: tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py (hunk 5 above)
Comment (skrsticTT) [reply tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py:88]: yes it can be harder to have different bfloat16 repr of higher values, but i mean this is going to be used just for debugging and in the almost every case you are going to have smaller tensor and you are going to debug just first core. do you agree?

Comment (nilaykp) [reply tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py:88]: higher dims was just for extreme case, even for small cases when you mulitply-add that with similar kernel tensor of say size 5X5 and add all channels, that would be quite a big number and values will start showing similarity too often(considering loss and 7 bit mantissa). However, I agree that this is for debug purpose only so will leave upto you.

**Review by nilaykp — APPROVED** [review]

File: tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py (hunk 4 above)
Comment (s-jovic) [reply tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py:96]: maybe we can add a common file in `tests/ttnn/nightly/unit_tests/operations`

Comment (skrsticTT) [reply tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py:96]: or maybe we can just import func from one to second file, not sure if it is ok to have this in common with other ops?

Comment (s-jovic) [reply tests/ttnn/nightly/unit_tests/operations/conv/test_conv2d.py:96]: you're right, this is specific to CNN ops.

either that or just to be explicit with the name of the utils/common file, sth like `cnn_utils.py.` your call anyway, it was a nit comment :)

**Review by s-jovic — APPROVED** [review]
